uniond/app/wasm.go

```diff
@@ -0,0 +1,176 @@
+package app
+
+import (
+	"fmt"
+	"path/filepath"
+
+	storetypes "cosmossdk.io/store/types"
+	"github.com/CosmWasm/wasmd/x/wasm"
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	wasmvm "github.com/CosmWasm/wasmvm/v2"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/posthandler"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibcfee "github.com/cosmos/ibc-go/v8/modules/apps/29-fee"
+	porttypes "github.com/cosmos/ibc-go/v8/modules/core/05-port/types"
+)
+
+// ContractMemoryLimit is the memory limit of each contract execution (in MiB)
+// constant value so all nodes run with the same limit.
+const ContractMemoryLimit = 32
+
+// registerWasmModules register CosmWasm keepers and non dependency inject modules.
+func (app *App) registerWasmModules(
+	appOpts servertypes.AppOptions,
+	wasmOpts ...wasmkeeper.Option,
+) (porttypes.IBCModule, error) {
+	// set up non depinject support modules store keys
+	if err := app.RegisterStores(
+		storetypes.NewKVStoreKey(wasmtypes.StoreKey),
+	); err != nil {
+		panic(err)
+	}
+
+	scopedWasmKeeper := app.CapabilityKeeper.ScopeToModule(wasmtypes.ModuleName)
+
+	nodeConfig, err := wasm.ReadNodeConfig(appOpts)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading wasm config: %s", err)
+	}
+
+	// create a new wasmvm with our own settings
+	wasmDir := filepath.Join(DefaultNodeHome, "wasm")
+	wasmer, err := wasmvm.NewVM(
+		wasmDir,
+		wasmkeeper.BuiltInCapabilities(),
+		ContractMemoryLimit,
+		nodeConfig.ContractDebugMode,
+		nodeConfig.MemoryCacheSize,
+	)
+	if err != nil {
+		panic(err)
+	}
+	wasmOpts = append(wasmOpts, wasmkeeper.WithWasmEngine(wasmer))
```

Comment:
Enforced overwriting on wasmd/wasmvm options, copied over from previous binary. Worth confirming this is necessary @hussein-aitlahcen